drivers/xen/events.c

```diff
@@ -227,12 +227,12 @@ static void events_isr(void *data)
 	while (pos_selector) {
 		/* Find first position, clear it in selector and process */
 		pos_index = __builtin_ffsl(pos_selector) - 1;
-		pos_selector &= ~(1 << pos_index);
+		pos_selector &= ~(((xen_ulong_t) 1) << pos_index);
```

Comment:
out of curiosity, why not using `1ULL` o `1UL` nstead?

Reply:
Such version will be more consistent with Xen approach. AFAIK definition of `xen_ulong_t` and structure, where events are stored various on different archs. So, `1UUL` will be OK for arm64, but probably will be a problem if someone will try to use it with x86 in future.

Reply:
> Such version will be more consistent with Xen approach. AFAIK definition of xen_ulong_t and structure, where events are stored various on different archs. So, 1UUL will be OK for arm64, but probably will be a problem if someone will try to use it with x86 in future.

So on x86, ``xen_ulong_t`` is effectively an ``unsigned long``. But this as some consequences because the 32-bit and 64-bit structure layout will be different and this requires a compat layer to support 32-bit guest on 64-bit hypervisor.

On Arm, we decided to go with ``uint64_t`` so the ABI doesn't change between Arm32 an Arm64.

At the moment, we technically use``1ULL``. However, I think is better to cast to the same type as the resulting type. This would help if someone decide to define ``xen_ulong_t`` as something bigger than ``unsigned long long``.